crates/lang/ir/src/ir/attrs.rs

```diff
@@ -549,16 +575,13 @@ impl InkAttribute {
     pub fn ensure_no_conflicts<'a, P>(
         &'a self,
         mut is_conflicting: P,
-    ) -> Result<(), syn::Error>
+    ) -> Result<(), &'a ir::AttributeArg>
```

Comment:
While I can understand the change behind this I am not sure I like an `ir::AttributeArg` as error return type.